pipelines/pathway.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package pipelines
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+	"log"
+	"math/rand"
+	"time"
+)
+
+// Pathway represents a path points can take.
+// It is defined as nodes (services) linked together with edges.
+// To reduce the size of the propagated serialized pathway, instead of storing
+// a list of edges and services, a hash of the path is computed. The hash is then resolved
+// in the Datadog backend.
```

Comment:
This is an implementation detail that does not belong in the public documentation. You could put this on top of the `hash` field if you like.